internal/builders/docker/common.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022 SLSA Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+// TODO(#1191): Update to the final BuildType URI.
+const DockerBasedBuildType = "https://slsa.dev/container-based-build/v0.1?draft"
+
+// BuildConfig is a collection of parameters to use for building the artifact.
+type BuildConfig struct {
+	// TODO(#1191): Add env and options if needed. 
+	// Command to pass to `docker run`. The command is taken as an array
+	// instead of a single string to avoid unnecessary parsing. See
+	// https://docs.docker.com/engine/reference/builder/#cmd and
+	// https://man7.org/linux/man-pages/man3/exec.3.html for more details.
+	Command []string `toml:"command"`
+	// The path, relative to the root of the git repository, where the artifact
+	// built by the `docker run` command is expected to be found.
+	ArtifactPath string `toml:"artifact_path"`
+}
+
+// BuildDefinition contains the infomraiton required for building an artifact using a Docker image.
+// Based on BuildDefinition in https://github.com/slsa-framework/slsa/pull/525. 
+type BuildDefinition struct {
```

Comment:
I think BuildDefinition/ParameterCollection/ArtifactReference might belong under `slsa/` sub-directory, since they're SLSA common definitions.  (maybe under a slsa/v1/common.go)?

Reply:
I agree. But I think eventually we want to use the structs/protobuf in https://github.com/slsa-framework/slsa/pull/525. For this PR I added these structs to be able to show-case the general idea of the builder... I can move them around. Let's see how this PR goes.

Reply:
SGTM -- let's see if these structs can hold the information needed to build and also be directly used for the provenance.